gazebo_plugins/include/gazebo_plugins/gazebo_ros_camera.hpp

```diff
@@ -84,15 +87,47 @@ class GazeboRosCamera : public gazebo::CameraPlugin
     unsigned int _width, unsigned int _height,
     unsigned int _depth, const std::string & _format) override;
 
+  /// Callback when depth camera produces a new image.
+  /*
+  * \details This is called at the camera's update rate.
+  * \details Not called when the camera isn't active. For a triggered camera, it will only be
+  * called after triggered.
+  * \param[in] _image Image
+  * \param[in] _width Image width
+  * \param[in] _height Image height
+  * \param[in] _depth Image depth
+  * \param[in] _format Image format
+  */
+  void OnNewImageFrame(
+    const unsigned char * _image,
+    unsigned int _width, unsigned int _height,
+    unsigned int _depth, const std::string & _format) override;
+
+  /// Callback when camera produces a new depth image.
+  /*
+  * \details This is called at the camera's update rate.
+  * \details Not called when the camera isn't active. For a triggered camera, it will only be
+  * called after triggered.
+  * \param[in] _image Image
+  * \param[in] _width Image width
+  * \param[in] _height Image height
+  * \param[in] _depth Image depth
+  * \param[in] _format Image format
+  */
+  void OnNewDepthFrame(
+    const float * _image,
+    unsigned int _width, unsigned int _height,
+    unsigned int _depth, const std::string & _format) override;
+
   /// Callback when camera is triggered.
-  void OnTrigger(const std_msgs::msg::Empty::SharedPtr _dummy);
+  void OnTrigger(std_msgs::msg::Empty::SharedPtr _dummy);
```

Comment:
I got a suggestion that 'const' declaration has no affect when used in function prototype, it only affects in the definition. Nevertheless I will revert it back

Reply:
Yeah it can be different in the two places. Having it in the prototype is good for documentation, so anyone looking at the header knows that the pointer is not going to be changed. For the `const bool` below it's really not doing anything, because the argument is passed by copy anyway.

In any case, I think removing `const` [breaks ABI](https://community.kde.org/Policies/Binary_Compatibility_Issues_With_C%2B%2B#The_Do.27s_and_Don.27ts), so we have to be careful about that.